server/lib/ReportManager.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const q = require('q');
+const path = require('path');
+const BadRequest = require('./errors/BadRequest');
+const supportedRender = {
+  json : require('./renderers/json'),
+  html : require('./renderers/html'),
+  pdf  : require('./renderers/pdf')
+};
+
+const defaultRender = 'pdf';
+const defaultOptions = { pageSize : 'A4', orientation: 'landscape' };
+const contentType = {
+  'pdf'  : '"Content-Type" : "application/pdf"',
+  'html' : '"Content-Type" : "application/html"',
+  'json' : '"Content-Type" : "application/json"'
+};
+
+// export the receipt object
+exports.build = build;
+
+/**
+ * @function build
+ * @desc build a report for inventory list of metadata
+ * @param {array} data inventory list of metadata
+ * @param {object} request the request which contains the renderer type (pdf, json, html, ...)
+ * @param {string} templateUrl the handlebars template url for the report
+ * @param {object} options the report page options
+ * @return {object} document the document blob
+ * @return {object} headers the document headers
+ * NOTE: the returned data are get by a spread(document, headers) function
+ * and the result is sended with res.set(headers).send(document)
+ */
+function build(data, req, templateUrl, options) {
```

Comment:
Since most `ExpressJS` handlers are `function (req, res, next)`, could you put `req` as the first parameter here?